test/performance/range/container_seq_write_benchmark.cpp

```diff
@@ -95,13 +95,64 @@ BENCHMARK_TEMPLATE(sequential_write, seqan3::bitcompressed_vector, seqan3::aa27)
 BENCHMARK_TEMPLATE(sequential_write, seqan3::bitcompressed_vector, seqan3::alphabet_variant<char, seqan3::dna4>);
 
 BENCHMARK_TEMPLATE(sequential_write, small_vec, char);
+BENCHMARK_TEMPLATE(sequential_write, small_vec, uint32_t);
 BENCHMARK_TEMPLATE(sequential_write, small_vec, seqan3::gap);
 BENCHMARK_TEMPLATE(sequential_write, small_vec, seqan3::dna4);
 BENCHMARK_TEMPLATE(sequential_write, small_vec, seqan3::gapped<seqan3::dna4>);
 BENCHMARK_TEMPLATE(sequential_write, small_vec, seqan3::dna15);
 BENCHMARK_TEMPLATE(sequential_write, small_vec, seqan3::aa27);
 BENCHMARK_TEMPLATE(sequential_write, small_vec, seqan3::alphabet_variant<char, seqan3::dna4>);
 
+// ============================================================================
+//  SeqAn2: sequential_write
+// ============================================================================
+
+#if SEQAN3_HAS_SEQAN2
+
+#include <seqan/sequence.h>
+
+template <template <typename...> typename container_t, typename spec_t, typename alphabet_t>
+void sequential_write2(benchmark::State & state)
+{
+    auto cont_rando = seqan3::test::generate_sequence_seqan2<alphabet_t>(10'000, 0, 0);
+    container_t<alphabet_t, spec_t> source = cont_rando;
+
+    alphabet_t a{};
+    for (auto _ : state)
+    {
+        for (auto && c : source)
```

Comment:
```suggestion
        for (auto & c : source)
```
Otherwise the assignment might not be valid if indeed a temporary is returned. Which is never the case but the compiler error would point to the actual issue

Reply:
> Otherwise the assignment might not be valid if indeed a temporary is returned. Which is never the case but the compiler error would point to the actual issue

Nono, we need this because some contaiers (e.g. bitcompressed) return proxies. The temporaries are materialised, that's the point of the proxy.